docs/writing-tests/making-a-testing-plan.md

```diff
@@ -0,0 +1,507 @@
+# Making a Testing Plan
+
+When contributing to a project as large and open-ended as WPT, it's easy to get
+lost in the details. Starting with an informal testing strategy helps you
+anticipate how much work will be involved, and it helps you stay focused once
+you begin.
+
+Many people come to WPT with a general testing goal in mind:
+
+- specification authors often want to test for new spec text
+- browser maintainers often want to test new features or fixes to existing
+  features
+- web developers often want to test discrepancies between browsers on their web
+  applications
+
+(If you don't have any particular goal, we can help you get started. Check out
+[the issues labeled with `type:missing-coverage` on
+GitHub.com](https://github.com/web-platform-tests/wpt/labels/type%3Amissing-coverage).
+Leave a comment if you'd like to get started with one, and don't hesitate to
+ask clarifying questions!)
+
+This guide will:
+
+1. show you how to use the specifications to learn what kinds of tests will be
+   most helpful
+2. help you develop a sense for what *doesn't* need to be tested
+3. demonstrate methods for figuring out which tests (if any) have already been
+   written for WPT
+
+## Understanding the "testing surface"
+
+Web platform specifications are instructions about how a feature should work.
+They're critical for implementers to "build the right thing," but they are also
+important for anyone writing tests. We can use the same instructions to infer
+what kinds of tests would be likely to detect mistakes. Here are a few common
+patterns in specification text and the kind of tests they suggest.
+
+### Input sources
+
+Algorithms may accept input from many sources. Modifying the input is the most
+direct way we can influence the browser's behavior and verify that it matches
+the specifications. That's why it's helpful to be able to recognize all the
+different sources of input.
+
+```eval_rst
+================ ==============================================================
+Type of feature  Potential input sources
+================ ==============================================================
+JavaScript       parameters, `context object <https://dom.spec.whatwg.org/#context-object>`_
```

Comment:
This code block is interpreted as ReStructured Text since Sphinx uses CommonMark, and CommonMark doesn't support tables. The trailing underscore is part of [the rST syntax for hyperlinks](http://docutils.sourceforge.net/docs/user/rst/quickref.html#hyperlink-targets).

Reply:
Ah, my bad, sorry!